server/dbversion.js

```diff
@@ -0,0 +1,74 @@
+const fs = require('fs')
+const path = require('path')
+const Umzug = require('umzug')
+const db = require('./models')
+
+// Sequelize uses umzug as the engine for running migrations, and stored completed migrations in a table
+// This module queries that table to see what migrations have been run against the database, and checks
+// against what migrations are available to this code. If there is a mismatch the throwAnyErrors method
+// will raise an error.
+// If sequelize stops using umzug in the future, umzug can be used to run the migrations directly so
+// we can have complete control of that part as well. We can also do this if it becomes necessary to
+// store things such as the date a migration took place, or was rolled back.
+
+class DbVersion {
+  constructor () {
+    this.availableVersions = []
+    this.highestVersion = ''
+    this.currentDatabaseVersion = ''
+    this.umzug = new Umzug({

```

Comment:
Not sure about instantiating Umzug within the DbVersion module. I've just discussed this briefly with Matthew and we agreed it warrants some thought.

I haven't put much thought into this yet, but my gut feeling is DbVersion should be moved to (or instantiated from) a Hapi plugin which would check the database version before running any route that depends on the database. This would ensure that no operation is ever executed against an incompatible database version.

Reply:
I don't think it needs to check before every route, as that could add a performance problem. It needs to deal gracefully with errors (such as database not saving due to schema or connection problems) and put the database version check in the readiness probe.